Lib/urlparse.py

```diff
@@ -205,18 +205,19 @@ def urlsplit(url, scheme='', allow_fragments=True):
     Note that we don't break the components up in smaller bits
     (e.g. netloc is a single string) and we don't expand % escapes."""
 
-    # Only lstrip url as some applications rely on preserving trailing space.
-    # (https://url.spec.whatwg.org/#concept-basic-url-parser would strip both)
-    url = url.lstrip(_WHATWG_C0_CONTROL_OR_SPACE)
-    scheme = scheme.strip(_WHATWG_C0_CONTROL_OR_SPACE)
-
     allow_fragments = bool(allow_fragments)
     key = url, scheme, allow_fragments, type(url), type(scheme)
     cached = _parse_cache.get(key, None)
     if cached:
         return cached
     if len(_parse_cache) >= MAX_CACHE_SIZE: # avoid runaway growth
         clear_cache()
+
```

Comment:
Why did this move?

Reply:
Because the new stripping of illegal characters (I believe, particularly NLs) caused tests to fail. I believe this configuration reduced the failures. The problem is, are you caching the one with newlines or without. I believe I moved it so that the NLs would be cached.

Reply:
I've run this by Jeremy, and he says okay.